packages/ozone/src/setting/service.ts

```diff
@@ -0,0 +1,146 @@
+import Database from '../db'
+import { Selectable } from 'kysely'
+import { Option } from '../lexicon/types/tools/ozone/setting/defs'
+import { paginate, TimeIdKeyset } from '../db/pagination'
+import { Setting, SettingScope } from '../db/schema/setting'
+import { Member } from '../db/schema/member'
+
+export type SettingServiceCreator = (db: Database) => SettingService
+
+export class SettingService {
+  constructor(public db: Database) {}
+
+  static creator() {
+    return (db: Database) => new SettingService(db)
+  }
+
+  async query({
+    limit = 100,
+    scope,
+    did,
+    cursor,
+    prefix,
+    keys,
+  }: {
+    limit: number
+    scope?: 'personal' | 'instance'
+    did?: string
+    cursor?: string
+    prefix?: string
+    keys?: string[]
+  }): Promise<{
+    options: Selectable<Setting>[]
+    cursor?: string
+  }> {
+    let builder = this.db.db.selectFrom('setting').selectAll()
+    const { ref } = this.db.db.dynamic
+
+    if (prefix) {
+      builder = builder.where('key', 'like', `${prefix}%`)
+    } else if (keys?.length) {
+      builder = builder.where('key', 'in', keys)
+    }
+
+    if (scope) {
+      builder = builder.where('scope', '=', scope)
+    }
+
+    if (did) {
+      builder = builder.where('did', '=', did)
+    }
+
+    const keyset = new TimeIdKeyset(ref(`setting.createdAt`), ref('setting.id'))
```

Comment:
Is it possible that `id` is enough for pagination?

Reply:
yeah that should work I think.